src/cases/e3sm_io_case_pio.hpp

```diff
@@ -99,7 +99,7 @@ inline int e3sm_io_pio_define_var (e3sm_io_driver &driver,
             // CHECK_ERR
 
             ibuf = var->decomp_id + 512;
-            err  = driver.put_att (fid, var->data, "__e3sm_io__/decomp", MPI_INT, 1, &ibuf);
+            err  = driver.put_att (fid, var->data, "__e3sm_io__/decomp", MPI_INT, 1, &ibuf, coll);
```

Comment:
This `driver.put_att` is called in folder `cases`, not `drivers`.
The last argument does not appear to be "within the driver."

Reply:
There must be a way to tell drivers that this call is to be collective or independent.
Driver can determine whether to follow or to proceed with collective call.

Since "\_\_pio\_\_" attributes is the behavior of the Scorpio module, not ADIOS2, it should be mimic by the blob test case, not the ADIOS driver.
The change made drivers able to take independent attribute call.
It should be up to the test cases to decide that it calls independent or collective.

Hard coding all "\_\_pio\_\_" prefixed attributes to independent call in the ADIOS2 driver is not portable and contradicts the purpose of driver interface.

Since we decide to implement the blob test as a test case, not a driver. It should not require special treatment in any of the drivers

Reply:
Hard coding it in ADIOS2 driver requires string comparison of all attributes to search for "\_\_pio\_\_" substring.
If we want to hard code, it is better to do it in the blob test case.
The impact will be that the blob test case can only be run by ADIOS2 driver.